staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator.go

```diff
@@ -207,14 +216,22 @@ func (s *specAggregator) buildOpenAPISpec() (specToReturn *spec.Swagger, err err
 
 // updateOpenAPISpec aggregates all OpenAPI specs.  It is not thread-safe. The caller is responsible to hold proper locks.
 func (s *specAggregator) updateOpenAPISpec() error {
-	if s.openAPIService == nil {
+	if s.openAPIService == nil || s.openAPIVersionedService == nil {
+		// openAPIVersionedService and deprecated openAPIService should be initialized together
+		if !(s.openAPIService == nil && s.openAPIVersionedService == nil) {
+			return fmt.Errorf("unexpected openapi service initialization error")
+		}
 		return nil
 	}
 	specToServe, err := s.buildOpenAPISpec()
 	if err != nil {
 		return err
 	}
-	return s.openAPIService.UpdateSpec(specToServe)
+	err = s.openAPIService.UpdateSpec(specToServe)
```

Comment:
I am pretty sure somewhere in aggregation, we access /swagger.json file. It should be somewhere in kube-openapi repo. Can you please verify that we updated that code to try version'ed endpoint first (with fallback to /swagger.json endpoint).

Reply:
I didn't find `/swagger.json` being accessed in kube-openapi repo. Do you mean `staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/downloader.go`? Change added